### PR TITLE
Questionnaire auto closing

### PR DESCRIPTION
**Pull request by TheBestTvarynka**
- [X] Request automatically closes when all users submit the questionnaire;
- [X] Request automatically closes when deadline expired;

### CODE REVIEW

File: backend/core/src/main/java/com/feed_grabber/core/response/ResponseService.java [hunk 1]
```diff
@@ -42,7 +51,35 @@ public ResponseService(ResponseRepository responseRepository) {
...
         response.setAnsweredAt(new Date());
 
-        return Optional.of(ResponseMapper.MAPPER.responseToDto(responseRepository.save(response)));
+        var result = Optional.of(ResponseMapper.MAPPER.responseToDto(responseRepository.save(response)));
+
+        int unanswered = responseRepository.countUnanswered(request.getId());
+        if (unanswered == 0) {
+            // close the request
+            requestService.closeNow(request.getId());
+            var toSaveNotification = UserNotification
```
Comment (MaksymSlobodianyk) [inline backend/core/src/main/java/com/feed_grabber/core/response/ResponseService.java:60]: We have new logic for notifications

**Review by MaksymSlobodianyk — CHANGES_REQUESTED** [review]

File: backend/core/src/main/java/com/feed_grabber/core/response/ResponseService.java (hunk 1 above)
Comment (MaksymSlobodianyk) [reply backend/core/src/main/java/com/feed_grabber/core/response/ResponseService.java:60]: It's better to have a call

File: backend/core/src/main/java/com/feed_grabber/core/request/RequestService.java [hunk 2]
```diff
@@ -193,6 +205,39 @@ public Date closeNow(UUID requestId) throws NotFoundException {
         return closeDate;
     }
 
+    public Date close(UUID requestId) throws NotFoundException {
+        var request = requestRepository.findById(requestId)
+                .orElseThrow(() -> new NotFoundException("Request not found"));
+        if (request.getCloseDate() != null) {
+            return new Date();
+        }
```
Comment (MaksymSlobodianyk) [inline backend/core/src/main/java/com/feed_grabber/core/request/RequestService.java]: Are you shure about  return new Date(); Maybe return request.getCloseDate()?

**Review by MaksymSlobodianyk — CHANGES_REQUESTED** [review]

File: backend/core/src/main/java/com/feed_grabber/core/request/RequestService.java (hunk 2 above)
Comment (TheBestTvarynka) [reply backend/core/src/main/java/com/feed_grabber/core/request/RequestService.java]: You right, thanks.

**Review by MaksymSlobodianyk — APPROVED** [review]

**Pull request merged**